### PR TITLE
Allow to build docker images offline (do not always use `docker build --pull`)

### PR DESCRIPTION
**Pull request by ereslibre**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Passing `KUBE_BUILD_PULL_LATEST_IMAGES=n` will disable building the artifacts
with `--pull`, so they can be built in an offline environment.

By default, `KUBE_BUILD_PULL_LATEST_IMAGES` is `y`, so the default behavior
is kept unchanged.

This allows to work on an offline environment when regenerating container images artifacts.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: build/lib/release.sh [hunk 1]
```diff
@@ -372,7 +373,11 @@ EOF
         if [[ "${base_image}" =~ busybox ]]; then
           echo "COPY nsswitch.conf /etc/" >> "${docker_file_path}"
         fi
-        "${DOCKER[@]}" build --pull -q -t "${docker_image_tag}" "${docker_build_path}" >/dev/null
+        local -a docker_build_opts=()
+        if [[ "${KUBE_BUILD_PULL_LATEST_IMAGES}" =~ [yY] ]]; then
```
Comment (cblecker) [inline build/lib/release.sh]: Can we get a comment here explaining the option?

**Review by cblecker — COMMENTED** [review]
I don't see any issue with this.

**Comment by cblecker** [comment]
/sig release
/priority backlog

/cc @ixdy 
Any reason for concern here, Jeff?

File: build/lib/release.sh (hunk 1 above)
Comment (ereslibre) [reply build/lib/release.sh]: Sure! I'll update the PR.

**Comment by ereslibre** [comment]
@cblecker Added the comment.

**Comment by ixdy** [comment]
I originally added this in https://github.com/kubernetes/kubernetes/pull/39695 because we were depending on things like `debian:jessie` or `busybox:latest`, so it was important to pull the latest image.

I think now we use explicit versions (and don't even use busybox at all in this flow), so we could even remove the `--pull`, but it's fine to leave it too.

/lgtm
/approve

**Pull request merged**